pybamm/models/full_battery_models/base_battery_model.py

```diff
@@ -398,11 +403,49 @@ def build_model(self):
     def set_thermal_submodel(self):
 
         if self.options["thermal"] is None:
-            thermal_submodel = pybamm.thermal.Isothermal(self.param)
+            if self.options["dimensionality"] == 0:
+                thermal_submodel = pybamm.thermal.Isothermal(self.param)
+            elif self.options["dimensionality"] == 1:
+                thermal_submodel = pybamm.thermal.current_collector.Isothermal2D(
```

Comment:
Should this be `Isothermal1plus1S` to be consistent with `Full1plus1D` and `Lumped1plus1D` ?

Reply:
yeah probably would be more consistent that way, but the model is still valid in full 2 or 3D were we to ever implement that. I'll change to Nplus1D for now though so it fits more clearly with the model structure